### PR TITLE
Prevent EjectOnDeath.

### LINKED ISSUE #7725: Some times an e1 ejects from bombing a vehicle
Here you can see a Rifle man spawning from the heavy tank husk.
openra game_2015_03_24_02_28_54_857

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9933338-some-times-an-e1-ejects-from-bombing-a-vehicle?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by phrohdoh**
Closes #7725.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
Seems useful even if the behaviour from #7725 is left as it is.

:+1:

File: OpenRA.Mods.Common/Traits/Demolishable.cs [hunk 1]
```diff
@@ -17,11 +17,26 @@ public class DemolishableInfo : IDemolishableInfo, ITraitInfo
 	{
 		public bool IsValidTarget(ActorInfo actorInfo, Actor saboteur) { return true; }
 
-		public object Create(ActorInitializer init) { return new Demolishable(); }
+		[Desc("If true and this actor has EjectOnDeath, no actor will be spawned.")]
+		public readonly bool PreventsEjectOnDeath = true;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Demolishable.cs]: Units spawning from demolished structures was an important (and frustrating) part of the commando missions in TD.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Demolishable.cs]: Okay, I will invert the default.

**Comment by obrakmann** [comment]
I'm not sure about the use case. If you need to change the yaml rules anyway to set this, why not just remove `EjectOnDeath` from the actor instead?

**Comment by Phrohdoh** [comment]
@obrakmann You may want structures to eject actors unless C4'd, which is exactly the issue this was created for.

**Comment by obrakmann** [comment]
Ah, I see. :+1: then, after the default has been reversed to preserve the current behaviour.

**Comment by Phrohdoh** [comment]
Updated the default to `false`.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/796bd90e43247c5a65b649a8c4b27ff28b682fb6...d0b14dddcbe79afc120ddc17807490e08824fedf)

**Pull request merged**